pandas/core/frame.py

```diff
@@ -1396,10 +1396,9 @@ def iterrows(self) -> Iterable[tuple[Hashable, Series]]:
         for k, v in zip(self.index, self.values):
             s = klass(v, index=columns, name=k).__finalize__(self)
             if using_cow and self._mgr.is_single_block:
-                s._mgr.blocks[0].refs = self._mgr.blocks[0].refs  # type: ignore[union-attr]  # noqa
-                s._mgr.blocks[0].refs.add_reference(  # type: ignore[union-attr]
-                    s._mgr.blocks[0]  # type: ignore[arg-type, union-attr]
-                )
+                # Argument 1 to "add_references" of "BaseArrayManager" has incompatible
+                # type "Union[BlockManager, ArrayManager]"; expected "BaseArrayManager"
```

Comment:
To be honest, this is kind of bullshit .. (and just noise to add as a comment) There is no easy way to let mypy understand such typing with a base class?

Reply:
Theoretically we could make mypy aware that using_copy_on_write() guards agains ArrayManager, but I am not that familiar with mypy :)

Removed.